Draft tube model is created

pandeysudan27 opened the pull request:
Two of the draft tube can be choosen, Conical diffuser and Moody spreading pipes. The svg figure in the documentation is not enlarged. I could not fixed it.

pandeysudan1 committed:
Starting to model the Draft Tube

This is the simple model of the draft tube.

pandeysudan1 committed:
Starting to model the Draft Tube

This is the simple model of the draft tube.

pandeysudan1 committed:
Draft tube test case created

pandeysudan1 committed:
Draft tube test case created

pandeysudan27 committed:
Two models of draft tube created.

Both conical diffuser and moody spreading pipes draft tubes models are created.

pandeysudan27 committed:
Two models of draft tube created.

Both conical diffuser and moody spreading pipes draft tubes models are created.

pandeysudan27 committed:
DraftTube is on pull request

A simple description of draft tubes are added for conical diffuser and moody spreading pipes.

pandeysudan27 committed:
DraftTube is on pull request

A simple description of draft tubes are added for conical diffuser and moody spreading pipes.

dietmarw commented on OpenHPL/Icons/DraftTube.mo:
```diff
@@ -0,0 +1,41 @@
+partial class DraftTube "Draft tube icon"
+  parameter Boolean vertical=false "Display vertical icon instead"
+    annotation(Dialog(group = "Icon"),
+    choices(checkBox = true));
```
What is the point of this unused parameter?

pandeysudan27 replied:
I do not know actually. I duplicated icon of pipe made by Luibomyr. 
(The mesh-grid coloring is also not supported in slanted icons like drafttube)

dietmarw replied:
Well I know what it is for in the Icon you copied it from (I implemented that) but it does no sense here. So be careful when copying over code blindly. You should always know what the code does before using it :smirk:

pandeysudan27 replied:
Thanks. I hope to not repeat again.

dietmarw commented on OpenHPL/Waterway/DraftTube.mo:
```diff
@@ -0,0 +1,146 @@
+<li><b>Conical diffuser:</b> It is the most well-know draft tube which has efficiency of around 90&percnt;  and mostly used for low head reaction turbines.</li>
+<li><b>Moody spreading draft tubes:</b> When conical diffuser length exceeds beyond its stability for high head reaction turbines, either a elbow type draft tube is used which has around 70&percnt; of efficiency. However, other choice is to use Moody spreading draft tube that has efficiency of around 80&percnt;. The construction and design of Moody spreading draft tube is daunting and time consum ...
+</ul>
+<p><br>The conical diffuser and Moody spreading draft tubes are shown below:</p>
```
```suggestion
<p>The conical diffuser and Moody spreading draft tubes are shown below:</p>
```

dietmarw commented on OpenHPL/Waterway/DraftTube.mo:
```diff
@@ -0,0 +1,146 @@
+<li><b>Moody spreading draft tubes:</b> When conical diffuser length exceeds beyond its stability for high head reaction turbines, either a elbow type draft tube is used which has around 70&percnt; of efficiency. However, other choice is to use Moody spreading draft tube that has efficiency of around 80&percnt;. The construction and design of Moody spreading draft tube is daunting and time consum ...
+</ul>
+<p><br>The conical diffuser and Moody spreading draft tubes are shown below:</p>
+<p><img src=\"modelica://OpenHPL/Resources/Images/conicalDiffuser.svg\"/></p>
```
```suggestion
<p><img src=\"modelica://OpenHPL/Resources/Images/conicalDiffuser.svg\" style=\"width:60%\"></p>
```

dietmarw commented on OpenHPL/Waterway/DraftTube.mo:
```diff
@@ -0,0 +1,146 @@
+</ul>
+<p><br>The conical diffuser and Moody spreading draft tubes are shown below:</p>
+<p><img src=\"modelica://OpenHPL/Resources/Images/conicalDiffuser.svg\"/></p>
+<p><img src=\"modelica://OpenHPL/Resources/Images/moodySpreadingDT.svg\"/></p>
```
```suggestion
<p><img src=\"modelica://OpenHPL/Resources/Images/moodySpreadingDT.svg\" style=\"width:60%\"></p>
```

dietmarw reviewed (commented):
I've left some comments and suggestions how to fix the scaling. You will only see this in OpenModelica since Dymola's HTML renderer does not support the svg scaling yet. Still I would not worry about that. Simply accept the suggestions by clicking on the "Commit suggestion" button.

pandeysudan27 committed:
Update OpenHPL/Waterway/DraftTube.mo

Co-Authored-By: Dietmar Winkler <dietmar.winkler@dwe.no>

pandeysudan27 committed:
Update OpenHPL/Waterway/DraftTube.mo

Co-Authored-By: Dietmar Winkler <dietmar.winkler@dwe.no>

pandeysudan27 committed:
Update OpenHPL/Waterway/DraftTube.mo

Co-Authored-By: Dietmar Winkler <dietmar.winkler@dwe.no>

pandeysudan27 committed:
Update OpenHPL/Waterway/DraftTube.mo

Co-Authored-By: Dietmar Winkler <dietmar.winkler@dwe.no>

pandeysudan27 committed:
Update OpenHPL/Waterway/DraftTube.mo

Co-Authored-By: Dietmar Winkler <dietmar.winkler@dwe.no>

pandeysudan27 committed:
Update OpenHPL/Waterway/DraftTube.mo

Co-Authored-By: Dietmar Winkler <dietmar.winkler@dwe.no>

pandeysudan27 committed:
Merge remote-tracking branch 'origin/draftTube' into draftTube

pandeysudan27 committed:
Typing error "MoodySpreadingPipes" instead of "MoodySpreadingPipe". 

I could not figure this out because it did work fine in my machine. strange!!

pandeysudan27 committed:
Data instance removed from HPSimple.mo

pandeysudan27 committed:
Physics of Moody pipes improved.

The mechanistic concept of Moody pipes is improved regarding mass, momentum and normal area that should be taken for pressure force.

dietmarw committed:
Fixing the initial flow rate for pipe and surge tank

Do not force the start flow rate but treat as guess value.

dietmarw committed:
Merge branch 'master' into draftTube

dietmarw commented:
Contained in #20
